pkg/apis/v1beta1/labels.go

```diff
@@ -0,0 +1,106 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"k8s.io/apimachinery/pkg/util/sets"
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis"
+)
+
+func init() {
+	karpv1.RestrictedLabelDomains = karpv1.RestrictedLabelDomains.Insert(RestrictedLabelDomains...)
+	karpv1.WellKnownLabels = karpv1.WellKnownLabels.Insert(
```

Comment:
scary we have 2 copies of this insert now (one in v1alpha, one here).

Is that an issue?

Reply:
As a matter of fact, is there anything wrong with removing all (or most) of v1alpha2, and just aliasing the types?

```go
type AKSNodeClass = v1beta1.AKSNodeClass
type AKSNodeClassList = v1beta1.AKSNodeClassList
type AKSNodeClassStatus = v1beta1.AKSNodeClassStatus
```

Reply:
I like the idea of aliasing the types, IF controller-gen reads the kubebuilder comments of aliased types correctly and correctly sets the storage version, etc.

Reply:
I think aliasing is a bit too cute. I have removed the duplication from v1alpha2 (and I agree we should be managing these standard labels and annotations in a more durable way, we can follow-up with that). For now, v1alpha2 deprecation + removing init duplication seems like the best bridge towards removing v1alpha2 altogether.

Wdyt?

Reply:
To me aliasing feels appropriate, given the stated intent of creating the exact copy, concerns around keeping them in sync, and concerns around accidental duplication of wiring. But I am good with any working solution, especially since we plan to remove it altogether in the near future. Let's get it in!